src/error.rs

```diff
@@ -79,6 +80,15 @@ pub enum SierraAssertError {
     Range { ranges: Box<(Range, Range)> },
 }
 
+#[derive(Error, Debug)]
+pub enum CompilerError {
+    #[error("BoundedInt value is out of range: {:?} not within [{:?}, {:?}[", value, range.0, range.1)]
```

Comment:
```suggestion
    #[error("BoundedInt value is out of range: {:?} not within [{:?}, {:?}]", value, range.0, range.1)]
```

Reply:
I did that because the upper range is exclusive and I wanted to highlight that on the error so that the user can understand if for example the problem is that the value is exactly equal to the upper range. It was the mathematical symbol of a range with the upper bound excluded, "[", do you still want to modify it to put a closing bracket "]"?

Reply:
I saw it with parentheses on college for open ranges. Maybe there's more than one notation. I thought it was a typo.

Reply:
(Note that I in fact used a closing parentheses, for the same reason you state, in the suggestion, rather than closing square bracket)